source/codegen/templates/service.cpp.mako

```diff
@@ -41,9 +41,13 @@ namespace ${config["namespace_component"]}_grpc {
 %   for custom_type in custom_types:
   void ${service_class_prefix}Service::Copy(const ${custom_type["name"]}& input, ${namespace_prefix}${custom_type["grpc_name"]}* output) 
   {
-%     for field in custom_type["fields"]: 
-    output->set_${field["grpc_name"]}(input.${field["name"]});
-%     endfor
+%for field in custom_type["fields"]: 
+    %if field["type"] == 'NIComplexNumber':
```

Comment:
Can you make this apply more genericly to this scenario instead of looking for a specific string for the `field["type"]`? I'm not sure on the exact python syntax but basically something like `%if field["type"] in [field["grpc_name"] for field in custom_type["fields"]]` should have the same behavior but work for similar scenarios going forward.

Reply:
Done.